core/src/main/c/share/bitmap_index_utils.cpp

```diff
@@ -53,6 +53,11 @@ int64_t find_latest_for_key(int64_t k,
         bool is_offset_in_mapped_area = tail.offset() + tail.memory_size() <= value_memory_size;
         bool is_inconsistent = !key.is_block_consistent || !is_offset_in_mapped_area;
         if (is_inconsistent) {
+            // first value block can be outside of file mapping as well
+            bool inside_mapped_area = inconsistent_tail.offset() + inconsistent_tail.memory_size() <= value_memory_size;
```

Comment:
does this affect Java version of index reader as well ?

Reply:
no, java is able to extend and remap memory.
this is c++ implementation specific problem